pycrires/pipeline.py

```diff
@@ -3884,6 +3884,140 @@ def correct_wavelengths(
+                        spatial_oversampling = 1.,
+                        util_extract_oversample = 12
+                        ) -> None:
+        """
+        Method for extracting spectra from the products of obs_nodding, while retaining
+        the spatial information.
+        Make sure that you have run obs_nodding first. 
+        The spectra are extracted by running util_extract over a range of slit fractions.
+
+        Parameters
+        ----------
+        nod_ab : str
+            Nod position with the spectra of which the wavelength
+            solution will be corrected.
+        verbose : bool
+            Print output produced by ``esorex``.
+        extraction_length: float
+            Spatial extent over which to extract the spectrum in arcseconds.
+            If 'None' then the pixel scale of the instrument is used.
+        spatial_oversampling: float
+            Oversampling factor for the extraction along the slit length.
+            Example: If spatial_oversampling=2 is being used the end result
+            will have twice as many spatial pixels as the original data.
+        util_extract_oversample: int
+            Oversampling to use for util_extract.
+        Returns
+        -------
+        NoneType
+            None
+        """
+
+        self._print_section("Process nodding frames", recipe_name="cr2res_util_extract_2d")
+
+        fits_files = sorted(list(self.file_dict[f'OBS_NODDING_COMBINED{nod_ab}'].keys()))
+
+        # Iterate over nod A exposures
+        for count_exp,fits_file in enumerate(fits_files):
+            
+            # Prepare SOF file
+            sof_file = pathlib.Path(self.product_folder / f"files_{count_exp:03d}.sof")
+            sof_open = open(sof_file, "w", encoding="utf-8")
+            sof_open.write(f"{fits_file} OBS_NODDING_OTHER\n")
+            tw_file = self.product_folder / f"cr2res_obs_nodding_trace_wave_{nod_ab}_{count_exp:03d}.fits"
+            sof_open.write(f"{tw_file} UTIL_WAVE_TW\n")
+            sof_open.close()
+
+            # Get data on size of the slit on the detector
+            tw_data = fits.open(tw_file)[1].data
+            slit_range = tw_data['SlitFraction'][0]
+            num_orders = tw_data['All'].shape[0]
+            tot_slit_fraction = slit_range[-1] - slit_range[0]
+            full_slit_length = 10 #arcsec
+            
+            # If extraction_length not specified, calculate the pixel scale
+            if extraction_length == None:
+                slit_fraction_pixels = np.mean(tw_data['Upper']-tw_data['Lower'])
+                pixel_scale = full_slit_length * tot_slit_fraction  / slit_fraction_pixels
+                extraction_length = pixel_scale
+
+            # Calculate the spatial size for the extraction and
+            # the number of spatial points
+            extraction_fraction = extraction_length / full_slit_length
+            N_points = spatial_oversampling * tot_slit_fraction / extraction_fraction
```

Comment:
It seems that `N_points` is the same as `spatial_oversampling` times `slit_fraction_pixels` (but please check), so this line can be simplified I think

Reply:
With the above change, I no longer calculate slit_fraction_pixels